test/unit/lib/plugins/aws/deploy/lib/checkForChanges.test.js

```diff
@@ -1421,23 +1141,28 @@ describe('test/unit/lib/plugins/aws/deploy/lib/checkForChanges.test.js', () => {
             headObject: async (params) => generateMatchingHeadObjectResponse(serverless, params),
           },
           CloudWatchLogs: {
-            describeSubscriptionFilters: {
-              // TODO: Ensure same ARN as lambda on which it is configured
+            deleteSubscriptionFilter: deleteStub,
+            describeSubscriptionFilters: async () => {
+              const functionName = serverless.service.getFunction('fn1').name;
+              return {
+                subscriptionFilters: [
+                  {
+                    // destinationArn `arn:{partition}:lambda:{region}:{accountId}:function:{functionName}`
+                    // filterName {stack name}-{logical id}-{random alphanumeric characters}
+                    filterName: 'teststack-Fn1LogsSubscriptionFilterCloudWatchLog1-xxxxx',
```

Comment:
It'll be nice to generate this name with help of `naming` utils. We have access to stack name via `naming.getStackName()`, we can get _logical id_ via `naming.getCloudWatchLogLogicalId()`

Reply:
I think I've come up with the expected expression here, let me know if you had something else in mind.